assets/quick-order-list.js

```diff
@@ -1,307 +1,280 @@
-
-        allVariants.forEach((variant) => {
-          const cartQty = parseInt(variant.dataset.cartQuantity);
-          if (cartQty > 0) {
-            hasVariantsInCart = true;
-            items[parseInt(variant.dataset.quantityVariantId)] = 0;
-          }
-        });
-
-        if (!hasVariantsInCart) {
-          this.classList.add('hidden');
-        }
-
-        this.actions = {
-          confirm: 'confirm',
-          remove: 'remove',
-          cancel: 'cancel',
-        };
-
-        this.addEventListener('click', (event) => {
-          event.preventDefault();
-          if (this.dataset.action === this.actions.confirm) {
-            this.toggleConfirmation(false, true);
-          } else if (this.dataset.action === this.actions.remove) {
-            this.quickOrderList.updateMultipleQty(items);
-            this.toggleConfirmation(true, false);
-          } else if (this.dataset.action === this.actions.cancel) {
-            this.toggleConfirmation(true, false);
-          }
-        });
-      }
-
-      toggleConfirmation(showConfirmation, showInfo) {
-        this.quickOrderList
-          .querySelector('.quick-order-list-total__confirmation')
-          .classList.toggle('hidden', showConfirmation);
-        this.quickOrderList.querySelector('.quick-order-list-total__info').classList.toggle('hidden', showInfo);
-      }
-    }
-  );
-}
-
 if (!customElements.get('quick-order-list')) {
   customElements.define(
     'quick-order-list',
     class QuickOrderList extends BulkAdd {
+      cartUpdateUnsubscriber = undefined;
+
       constructor() {
         super();
-        this.cart = document.querySelector('cart-drawer');
-        this.quickOrderListId = `${this.dataset.section}-${this.dataset.productId}`;
-        this.defineInputsAndQuickOrderTable();
-
-        this.variantItemStatusElement = document.getElementById('shopping-cart-variant-item-status');
-        const form = this.querySelector('form');
-        this.inputFieldHeight = this.querySelector('.variant-item__quantity-wrapper').offsetHeight;
-        this.isListInsideModal = document.querySelector('.quick-add-bulk');
-        this.stickyHeaderElement = document.querySelector('sticky-header');
-        this.getTableHead();
-        this.getTotalBar();
+        this.isListInsideModal = this.closest('bulk-modal');
 
+        this.stickyHeaderElement = document.querySelector('sticky-header');
         if (this.stickyHeaderElement) {
           this.stickyHeader = {
             height: this.stickyHeaderElement.offsetHeight,
             type: `${this.stickyHeaderElement.getAttribute('data-sticky-type')}`,
           };
         }
 
-        if (this.getTotalBar()) {
-          this.totalBarPosition = window.innerHeight - this.getTotalBar().offsetHeight;
+        this.totalBar = this.getTotalBar();
+        if (this.totalBar) {
+          this.totalBarPosition = window.innerHeight - this.totalBar.offsetHeight;
 
           window.addEventListener('resize', () => {
```

Comment:
Pre-existing issue, but we should probably return the event listener here so we can unsubscribe from the resize event when the quick order list is unmounted/detached.